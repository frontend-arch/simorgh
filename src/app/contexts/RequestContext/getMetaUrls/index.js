const getAmpUrl = url => `${url}.amp`;

const getCanonicalUrl = (origin, pathname) => {
  const ampRegex = /(.amp)/g;
  const canonicalUrl = `${origin}${pathname}`;

  return canonicalUrl.replace(ampRegex, '');
};

const getUkCanonicalUrl = url => {
  const tldRegex = /(.com|.co.uk)/g;
  return url.replace(tldRegex, '.co.uk');
};

const getNonUkCanonicalUrl = url => {
  const tldRegex = /(.com|.co.uk)/g;
  return url.replace(tldRegex, '.com');
};

const getMetaUrls = (origin, pathname) => {
  const canonicalLink = getCanonicalUrl(origin, pathname);
  const canonicalUkLink = getUkCanonicalUrl(canonicalLink);
  const canonicalNonUkLink = getNonUkCanonicalUrl(canonicalLink);

  return {
    canonicalLink,
    ampLink: getAmpUrl(canonicalLink),
    canonicalUkLink,
    ampUkLink: getAmpUrl(canonicalUkLink),
    canonicalNonUkLink,
    ampNonUkLink: getAmpUrl(canonicalNonUkLink),
  };
};

export default getMetaUrls;
