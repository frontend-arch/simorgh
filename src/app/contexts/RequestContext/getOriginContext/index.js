const getOriginContext = bbcOrigin => {
  let origin = 'https://www.bbc.co.uk';
  let isUK = true;

  if (bbcOrigin) {
    origin = bbcOrigin;
  } else if (process && process.env && process.env.APP_ENV === 'local') {
    origin = process.env.SIMORGH_BASE_URL;
  } else if (
    typeof window !== 'undefined' &&
    window.location &&
    window.location.origin
  ) {
    origin = window.location.origin; // eslint-disable-line prefer-destructuring
  }

  if (origin.includes('.com')) {
    isUK = false;
  }

  return {
    origin,
    isUK,
  };
};

export default getOriginContext;
