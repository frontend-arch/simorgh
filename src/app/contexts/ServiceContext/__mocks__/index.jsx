import React from 'react';
import { node, string } from 'prop-types';
import services from '#server/utilities/serviceConfigs';

/*
 * This file is mocked by default to avoid having to handle
 * async behavior in tests across the application when
 * using service contexts.
 */
export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({ children, service, variant }) => (
  <ServiceContext.Provider value={services[service][variant || 'default']}>
    {children}
  </ServiceContext.Provider>
);

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string.isRequired,
  variant: string,
};

ServiceContextProvider.defaultProps = {
  variant: 'default',
};
