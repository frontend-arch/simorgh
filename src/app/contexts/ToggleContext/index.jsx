import React, { createContext, useReducer } from 'react';
import { node } from 'prop-types';
import toggleReducer from '../../reducers/ToggleReducer';
import defaultToggles from '#lib/config/toggles';

const ToggleContext = createContext({});

const ToggleContextProvider = ({ children }) => {
  const [toggleState, toggleDispatch] = useReducer(
    toggleReducer,
    defaultToggles,
  );

  return (
    <ToggleContext.Provider value={{ toggleState, toggleDispatch }}>
      {children}
    </ToggleContext.Provider>
  );
};

const ToggleContextConsumer = ToggleContext.Consumer;

ToggleContextProvider.propTypes = {
  children: node.isRequired,
};

export { ToggleContext, ToggleContextProvider, ToggleContextConsumer };
