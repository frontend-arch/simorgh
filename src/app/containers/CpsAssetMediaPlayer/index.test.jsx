import React from 'react';
import {
  shouldMatchSnapshot,
  isNull,
  suppressPropWarnings,
} from '@bbc/psammead-test-helpers';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';
import CpsAssetMediaPlayerContainer from '.';
import {
  validAresMediaVideoBlock,
  defaultToggles,
} from '../MediaPlayer/fixtureData';

const GenerateMediaPlayer = ({
  /* eslint-disable react/prop-types */
  platform,
  blocks,
  assetUri,
  /* eslint-enable react/prop-types */
}) => (
  <RequestContextProvider
    isAmp={platform === 'amp'}
    service="news"
    statusCode={200}
    platform={platform}
    id="foo"
    pageType="MAP"
    pathname="/pathname"
  >
    <ServiceContextProvider service="news">
      <ToggleContext.Provider
        value={{ toggleState: defaultToggles, toggleDispatch: jest.fn() }}
      >
        <CpsAssetMediaPlayerContainer blocks={blocks} assetUri={assetUri} />
      </ToggleContext.Provider>
    </ServiceContextProvider>
  </RequestContextProvider>
);

describe('MediaPlayer', () => {
  shouldMatchSnapshot(
    'render the canonical player without a placeholder',
    <GenerateMediaPlayer
      platform="canonical"
      blocks={[validAresMediaVideoBlock]}
      assetUri="/pidgin/123456789"
    />,
  );

  shouldMatchSnapshot(
    'render the amp player',
    <GenerateMediaPlayer
      platform="amp"
      blocks={[validAresMediaVideoBlock]}
      assetUri="/pidgin/123456789"
    />,
  );

  suppressPropWarnings(['assetUri']);
  isNull(
    'is Null when assetUri is not provided',
    <GenerateMediaPlayer
      platform="canonical"
      blocks={[validAresMediaVideoBlock]}
    />,
  );
});
