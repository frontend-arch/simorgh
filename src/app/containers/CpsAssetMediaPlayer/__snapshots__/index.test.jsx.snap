// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPlayer render the amp player 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    .c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c0 {
  margin-bottom: 1.5rem;
}

@media (max-width:63rem) {
  .c0 {
    grid-column: 1 / span 6;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-column: 3 / span 6;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-column: 6 / span 12;
  }
}

@media (max-width:25rem) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c0 {
    padding: 0 1rem;
  }
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <amp-iframe
            allowfullscreen="allowfullscreen"
            frameborder="0"
            layout="fill"
            sandbox="allow-scripts allow-same-origin"
            src="https://www.test.bbc.com/ws/av-embeds/cps/pidgin/123456789/p01k6msp/en-GB/amp"
            title="Media player"
          >
            <amp-img
              alt=""
              attribution=""
              height="16"
              layout="fill"
              placeholder="true"
              src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
              srcset="https://ichef.test.bbci.co.uk/images/ic/240xn/p01k6mtv.jpg 240w, https://ichef.test.bbci.co.uk/images/ic/320xn/p01k6mtv.jpg 320w, https://ichef.test.bbci.co.uk/images/ic/480xn/p01k6mtv.jpg 480w, https://ichef.test.bbci.co.uk/images/ic/624xn/p01k6mtv.jpg 624w, https://ichef.test.bbci.co.uk/images/ic/800xn/p01k6mtv.jpg 800w"
              width="9"
            />
          </amp-iframe>
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer render the canonical player without a placeholder 1`] = `
.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 0;
  overflow: hidden;
}

.c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c0 {
  margin-bottom: 1.5rem;
}

@media (max-width:63rem) {
  .c0 {
    grid-column: 1 / span 6;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-column: 3 / span 6;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-column: 6 / span 12;
  }
}

@media (max-width:25rem) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c0 {
    padding: 0 1rem;
  }
}

<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
  </head>
  <body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <iframe
            allow="autoplay; fullscreen"
            allowfullscreen=""
            class="c2"
            src="https://www.test.bbc.com/ws/av-embeds/cps/pidgin/123456789/p01k6msp/en-GB"
            title="Media player"
          />
        </div>
      </div>
    </div>
  </body>
</html>
`;
